.github/workflows/validate.yml

```diff
@@ -20,7 +20,8 @@ jobs:
         fetch-depth: 0
 
     - name: Setup dotnet
-      uses: actions/setup-dotnet@v4
+      # Using @main for https://github.com/actions/setup-dotnet/issues/645. Revert to @vN once fix is released.
+      uses: actions/setup-dotnet@main
```

Comment:
If we merge #2864 first, then we can revert this now. I will do an end-to-end build-and-test of internal repo with both changes before merging that.